### PR TITLE
Unify python plugin

### PR DESCRIPTION
**Pull request by SamYaple**

### CODE REVIEW

File: snapcraft/plugins/python.py [hunk 1]
```diff
@@ -45,14 +45,15 @@
       The python version to use. Valid options are: python2 and python3
```
Comment (SamYaple) [inline snapcraft/plugins/python.py]: ignore this whole commit, its only here to allow me to work on, and test, this change without an ugly rebase when this patch merges

**Comment by SamYaple** [comment]
I think it is funny and want to point this out: https://coveralls.io/builds/7852962

All edited files are 100% coveraged covered, but the overall percentage of coverage has fallen. This check would pass if snapcraft was 100% coveraged. thats pretty funny to me

**Comment by sergiusens** [comment]
It is funny, I am not a fan and we can ask @elopio (and he will say its to find dead test code most likely). But the fact that you unified tests means the total lines of code covered has reduced making the 100% mark harder to reach while still having the same amount of untested code.

**Comment by SamYaple** [comment]
Eh, I could always add in some filler code to make up some more lines...... :)

I am going to be working on a few more tests for the plugin in the future. Do proper unit tests against each function and mock all the rest. As it is right now, we are doing _ok_ with the unit tests, but I feel they are more busy work to get coveralls to pass rather than useful.

I don't plan on doing that for this commit though.

Once https://github.com/snapcore/snapcraft/pull/785 merges, I will rebase this and we can discuss what to do here.

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by SamYaple** [comment]
@elopio Over all coverage percentage has dropped due to the deduplication of the test suites, while code coverage has actually risen (in this case, they python plugins are now 100% coveraged). I believe that means we need to force merge this and ignore coveralls.

Thoughts?

**Comment by sergiusens** [comment]
oh, I intend to ignore it indeed!

File: snapcraft/tests/test_plugin_python.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
```
Comment (sergiusens) [inline snapcraft/tests/test_plugin_python.py]: date needs fixing here

**Comment by SamYaple** [comment]
Heh. ok.

Well the plugin is coming along quite nicely I think. It has (i believe) every feature I want now. I am goign to iron out the test suite a bit more and refactor a few sections, but functionally its solid for me now.

File: snapcraft/tests/test_plugin_python.py (hunk 2 above)
Comment (SamYaple) [inline snapcraft/tests/test_plugin_python.py]: how do you want it fixed? 2015-2016, or 2015, 2016

Comment (sergiusens) [inline snapcraft/tests/test_plugin_python.py]: I personally prefer 2015-2016 as then it is just a one digit change to maintain, but I don't mind either.

Comment (SamYaple) [inline snapcraft/tests/test_plugin_python.py]: i do as well. it shall be so

**Comment by sergiusens** [comment]
Is there a bug for this?

**Comment by SamYaple** [comment]
Not that I know of. Is that a hard requirement?

**Comment by sergiusens** [comment]
LP: #1623289

**Comment by SamYaple** [comment]
thanks

**Pull request merged**